### ISSUE TITLE
Update to public release 1057

### ISSUE THREAD

**Issue by bjakja** [haiyang830/KaiNote-language-zh_CN#1]
If you can update this translation to newest release, I can fix Kainote to load it as Chinese without changing name.

**Comment by haiyang830** [comment]
已经更新到KaiNote(v0.9.5.1057)的最新版本了
I have updated zh_CN.po to latest v0.9.5.1057

**Comment by bjakja** [comment]
Next version should work without renaming anything but without change texts in code after changes you will see Polish instead of English. And this will plenty of work change 1500 or more texts.

**Comment by SerCom-KC** [comment]
> but without change texts in code after changes you will see Polish instead of English

@bjakja Do you mean untranslated strings will fallback to Polish? This shouldn't be a problem, since copying strings from English as a temporary placeholder is also an acceptable option (depends on @haiyang830 though)

**Comment by haiyang830** [comment]
Thank you for your reminder, it is a hard job for Kainote to develop.

**Comment by bjakja** [comment]
> @bjakja Do you mean untranslated strings will fallback to Polish?
When I add new texts to program, you will see Polish but only in other versions then this 1057 build. 
![obraz](https://user-images.githubusercontent.com/15655326/49590454-3ec9d900-f96c-11e8-9c56-ee9022bcd5c7.png)
I will change it to English, in about 2-6 months. Till now most users of Kainote was from Poland and there was no need that I changed texts to English.
When I fix locale, I write it here. For now I have to rewrite Video player, and add original CSRI for compatibility with other subtitle filters.

**Comment by SerCom-KC** [comment]
> When I add new texts to program, you will see Polish but only in other versions then this 1057 build.

Yeah, I noticed that Kainote is heavily based on Polish at the moment. That shouldn't be a serious problem, but still thanks for the reminder!

**Comment by bjakja** [comment]
There is no other internationalization problem with Kainote?
Probably Chinese is shown in small text field in top left corner like Japanese.

**Comment by SerCom-KC** [comment]
@bjakja There are some word-ordering issues though:
https://github.com/haiyang830/KaiNote-language-zh_CN/blob/661476eaf5de716ec1b8dfa4bd6e1cf4d93f3443/zh_CN.po#L3283

In Chinese we prefer to translate it as `Before %s would you like to save subtitle as %s` to make it less awkward. We have a workaround for such lines at the moment, but it's hard to say if other languages can do the same.

**Comment by bjakja** [comment]
It looks like the other languages can have similar issues. It would be better to write three full versions of this text. Many of this text was written when I even not thought of adding English language. And some texts are split on more small texts that prevent to translate it properly. By internationalization issues I mean for example problems with writing some characters, bad working for example spellchecker etc. Text field is written by me, and I without any feedback cannot for example write it for right to left writing etc. Text field from Winapi do not have probably that issues but had others that prevent me from use it for example inability to colorize all text field with scrollbar.

**Comment by SerCom-KC** [comment]
> problems with writing some characters

If you mean displaying issues caused by bad [characters encoding](https://pl.wikipedia.org/wiki/Zestaw_znaków), no. Every character displays correctly so far.

> Probably Chinese is shown in small text field in top left corner like Japanese.

If you mean [Furigana](https://en.wikipedia.org/wiki/Furigana), no, we don't use that in Chinese.

> spellchecker

Most of us don't use spellchecker so we haven't tested on that yet.

> write it for right to left writing

We are also Left To Right so not a problem for us.

> inability to colorize all text field with scrollbar

Not sure what this is... Screenshot explanation please?

**Comment by bjakja** [comment]
> Not sure what this is... Screenshot explanation please?

I made own themes that I cannot use winapi text field or scintilla cause I cannot colorize scrollbar like it is actually in Kainote. Most windows of Kainote looks the same on every Windows OS. I made themes when after changing system theme and sending a screenshots, peoples want to have that dark Kainote.

**Comment by bjakja** [comment]
I thought that I can add translations for languages with names of language in each language like on screen below. 
![obraz](https://user-images.githubusercontent.com/15655326/49947178-e016d900-fef0-11e8-9d7b-cacf9b68027d.png)

**Comment by bjakja** [comment]
I will make a public release, could you update translation?
Mo file is updated on Github. 
I want to release it cause after upgrade to Windows 10 Kainote has some new graphical bugs. I have to change GDI that sux on Win10 to GDI+. It should make program graphics much faster, and can make many bugs. Debugging will take some days that I don't release it without proper testing.
